include/d/a/npc/d_a_npc.h

```diff
@@ -2,11 +2,228 @@
 #define D_A_NPC_H
 
 #include "common.h"
+#include "d/a/d_a_base.h"
 #include "d/a/d_a_itembase.h"
 #include "d/a/obj/d_a_obj_base.h"
+#include "d/d_message.h"
+#include "libms/flowfile.h"
+#include "m/m3d/m_mdl.h"
+#include "m/m_angle.h"
+#include "m/m_mtx.h"
+#include "m/m_quat.h"
 #include "m/m_vec.h"
 #include "s/s_State.hpp"
 #include "s/s_StateInterfaces.hpp"
+#include "toBeSorted/d_d3d.h"
+
+class dAcNpc_c;
+
+// NPC-specific mdl callback stuff
+class dNpcMdlCallbackBase_c {
+protected:
+    /* 0x00 */ u32 mNodeId;
+    /* 0x04 */ mAng field_0x04;
+    /* 0x06 */ mAng field_0x06;
+    /* 0x08 */ bool mForceCalc;
+
+    dNpcMdlCallbackBase_c() : mNodeId(-1), field_0x04(0), field_0x06(0), mForceCalc(false) {}
+
+public:
+    u32 getNodeId() const {
+        return mNodeId;
+    }
+
+    void loadNodeId(d3d::AnmMdlWrapper *mdl, const char *nodeName);
+
+    // vtable at 0xC
+    /* 0x08 */ virtual void apply(mMtx_c *result) const = 0;
+    /* 0x0C */ virtual void calc() = 0;
+    /* 0x10 */ virtual bool isDone() = 0;
+    /* 0x14 */ virtual void timingA(nw4r::g3d::ChrAnmResult *result);
+    /* 0x18 */ virtual void timingB(mMtx_c *result);
+};
+
+/**
+ * A mdl callback commonly used for controlling NPC head movement.
+ * NPCs will look at Link or certain objects of interest (e.g. a Bomb held by Link).
+ * This class smoothly interpolates the additive rotation of its node (head, neck, spine).
+ */
+class dNpcMdlCallbackAng_c : public dNpcMdlCallbackBase_c {
+protected:
+    /* 0x10 */ mAng3_c mTarget;
+    /* 0x16 */ s16 mNumSteps;
+    /* 0x18 */ mAng mMaxStepSizeYZ;
+    /* 0x1A */ mAng mMinStepSizeYZ;
+    /* 0x1C */ mAng mMaxStepSizeX;
+    /* 0x1E */ mAng mMinStepSizeX;
+    /* 0x20 */ s32 mScaleZ;
+    /* 0x24 */ s32 mScaleX;
+    /* 0x28 */ s32 mScaleY;
+    /* 0x2C */ mAng3_c mCurrent;
+
+    s32 getScaleX() const {
+        return mScaleX;
+    }
+
+    s32 getScaleY() const {
+        return mScaleY;
+    }
+
+    s32 getScaleZ() const {
+        return mScaleZ;
+    }
+
+public:
+    dNpcMdlCallbackAng_c();
+    /* 0x1C */ virtual ~dNpcMdlCallbackAng_c() {}
+
+    void setMinMaxStepSize(const mAng &max, const mAng &min);
+    void setMinMaxStepSizeX(const mAng &max, const mAng &min);
+    void reset();
+    void resetTarget();
+    void setTarget(const mAng3_c &target);
+    void setTargetNow(const mAng3_c &target);
+    void finishTarget();
+
+    /* 0x0C */ virtual void calc() override;
+    /* 0x10 */ virtual bool isDone() override {
+        // TODO - this matches, but maybe an operator overload?
+        return mCurrent.x == mTarget.x && mCurrent.y == mTarget.y && mCurrent.z == mTarget.z;
```

Comment:
A quick test and an operator overload works. Unsure if it regresses any other areas.